src/tensor_ops/ema/mod.rs

```diff
@@ -0,0 +1,116 @@
+use crate::{
+    shapes::{Shape, Unit},
+    tensor::{DeviceStorage, Tensor},
+};
+
+mod cpu_kernel;
+#[cfg(feature = "cuda")]
+mod cuda_kernel;
+
+/// Exponential moving average: `dst * decay + src * (1 - decay)`. See [ema_assign]
+/// for in place version.
+pub fn ema<S: Shape, E: Unit, D>(
```

Comment:
Should change this to be a more general: `c = alpha * a + beta * b`.

Then this could actually be used for ema (alpha = decay, beta = 1 - decay), and gradient accum (alpha = 1, beta = 1).